src/main/java/gregtech/api/worldgen/config/WorldGenRegistry.java

```diff
@@ -184,6 +214,21 @@ private static void extractJarVeinDefinitions(Path worldgenRootPath) throws IOEx
         }
     }
 
+    private void gatherNamedDimensions(Path dimensionsFile) {
+        try (InputStream fileStream = Files.newInputStream(dimensionsFile)) {
+            InputStreamReader streamReader = new InputStreamReader(fileStream);
+            JsonObject element = jsonParser.parse(streamReader).getAsJsonObject();
+            JsonArray dims = element.getAsJsonArray("dims");
+            for(JsonElement dim : dims) {
+                namedDimensions.put(dim.getAsJsonObject().get("dimName").getAsString(), dim.getAsJsonObject().get("dimID").getAsInt());
+            }
+        }
+        catch (IOException exception) {
+            GTLog.logger.error("Failed to load dimensions definition file on path {}", dimensionsFile, exception);
+        }
```

Comment:
This would just replicate the existing behavior of `extractJarVeinDefinitions` which takes the path, and parses the dimension or ore vein files if they don't exist. If they do exist, the extraction is done in `reinitializeRegisteredVeins`

Reply:
I have commit ready for this extraction as I mentioned. If you are interested